src/app/info/AddWikiItemButton.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import { DbWikiRow } from "@/databaseUtils";
+import { ButtonMargin } from "@/app/info/StyleComponents";
+import { Button } from "@mui/material";
+import supabase from "@/supabaseClient";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { PostgrestError } from "@supabase/supabase-js";
+
+interface WikiRowQuerySuccessType {
+    data: DbWikiRow;
+    error: null;
+}
+
+interface WikiRowQueryFailureType {
+    data: null;
+    error: PostgrestError;
+}
+export type WikiRowQueryType = WikiRowQuerySuccessType | WikiRowQueryFailureType;
+
+interface AddWikiItemButtonProps {
+    sortedRows: DbWikiRow[];
+    setSortedRows: (rows: DbWikiRow[]) => void;
+}
+
+const AddWikiItemButton: React.FC<AddWikiItemButtonProps> = ({ sortedRows, setSortedRows }) => {
+    const addWikiItem = async (): Promise<void> => {
+        if (
+            sortedRows.filter((row) => {
+                return !row.title && !row.content;
+            }).length === 0
+        ) {
```

Comment:
If this is the only thing we need sorted rows for, can we calculate that in the parent rather than passing the whole array through?